nixos/modules/services/web-servers/caddy/default.nix

```diff
@@ -303,13 +326,16 @@ in
       wantedBy = [ "multi-user.target" ];
       startLimitIntervalSec = 14400;
       startLimitBurst = 10;
+      reloadTriggers = optional cfg.enableReload configFile;
 
-      serviceConfig = {
+      serviceConfig = let
+        validateCommand = ''${cfg.package}/bin/caddy validate --config ${configPath} ${optionalString (cfg.adapter != null) "--adapter ${cfg.adapter}"}'';
+      in {
         # https://www.freedesktop.org/software/systemd/man/systemd.service.html#ExecStart=
         # If the empty string is assigned to this option, the list of commands to start is reset, prior assignments of this option will have no effect.
-        ExecStart = [ "" ''${cfg.package}/bin/caddy run --config ${cfg.configFile} ${optionalString (cfg.adapter != null) "--adapter ${cfg.adapter}"} ${optionalString cfg.resume "--resume"}'' ];
-        ExecReload = [ "" ''${cfg.package}/bin/caddy reload --config ${cfg.configFile} ${optionalString (cfg.adapter != null) "--adapter ${cfg.adapter}"} --force'' ];
-        ExecStartPre = ''${cfg.package}/bin/caddy validate --config ${cfg.configFile} ${optionalString (cfg.adapter != null) "--adapter ${cfg.adapter}"}'';
+        ExecStart = [ "" ''${cfg.package}/bin/caddy run --config ${configPath} ${optionalString (cfg.adapter != null) "--adapter ${cfg.adapter}"} ${optionalString cfg.resume "--resume"}'' ];
+        ExecReload = [ "" validateCommand ''${cfg.package}/bin/caddy reload --config ${configPath} ${optionalString (cfg.adapter != null) "--adapter ${cfg.adapter}"} --force'' ];
```

Comment:
Some comment on why `validateCommand` is added to `ExecReload` would be good.

Reply:
Why do we need that `validateCommand` at all?
Both `caddy reload` and `caddy run` will print "proper" errors, when an invalid config is passed.

An invalid config passed to a running instance via `caddy reload` will get rejected and that instance continues to use the previous config.
There is no risk associated with it.

Reply:
Sorry... can't recall the details. Was a while back 😅